src/App.jsx

```diff
@@ -51,7 +56,10 @@ export function App() {
 		<Router>
 			<Routes>
 				<Route path="/" element={<Layout />}>
-					<Route index element={<Home />} />
+					<Route
+						index
+						element={<Home onClick={handleClick} listToken={listToken} />}
+					/>
```

Comment:
you can redirect the user directly from here, with the `<Navigate />` component, without passing the `listToken` to the home page. `element={listToken ? <Navigate to="/list" /> :  <Home onClick={handleClick} />}` 
- Note: This is a one way to fix the flash, maybe there are other solutions.